UnityProject/Assets/Scripts/Transform/CustomNetTransform.Move.cs

```diff
@@ -0,0 +1,708 @@
+{
+
+	private const string STOPPED_FLOATING = "{0} stopped floating";
+	private const string PREDICTIVE_STOP_TO = "{0}: predictive stop @ {1} to {2}";
+	private const string NUDGE = "Nudge:{0} {1}";
+	private PushPull pushPull;
+	public PushPull PushPull => pushPull ? pushPull : (pushPull = GetComponent<PushPull>());
+
+	/// Containers and other objects meant to be snapped by tile
+	public bool IsTileSnap => registerTile.ObjectType == ObjectType.Object;
+
+	public bool IsClientLerping => transform.localPosition != MatrixManager.WorldToLocal(predictedState.WorldPosition, MatrixManager.Get(matrix));
+	public bool IsServerLerping => serverLerpState.WorldPosition != serverState.WorldPosition;
+	public bool CanPredictPush => !IsClientLerping;
+	public bool IsMovingClient => IsClientLerping;
+	public bool IsMovingServer => IsServerLerping;
+	public Vector2 ServerImpulse => serverState.WorldImpulse;
+	public float SpeedServer => ServerState.speed;
+	public float SpeedClient => PredictedState.speed;
+	public bool IsFloatingServer => serverState.WorldImpulse != Vector2.zero && serverState.Speed > 0f && !IsBeingPulledServer;
+	public bool IsFloatingClient => predictedState.WorldImpulse != Vector2.zero && predictedState.Speed > 0f && !IsBeingPulledClient;
+	public bool IsBeingThrownServer => !serverState.ActiveThrow.Equals(ThrowInfo.NoThrow);
+	public bool IsBeingThrownClient => !clientState.ActiveThrow.Equals(ThrowInfo.NoThrow);
+	public bool IsBeingPulledServer => pushPull && pushPull.IsBeingPulled;
+	public bool IsBeingPulledClient => pushPull && pushPull.IsBeingPulledClient;
+
+	/// (Server) Did the flying item reach the planned landing point?
+	private bool ShouldStopThrow
+	{
+		get
+		{
+			if (!IsBeingThrownServer && !IsBeingThrownClient)
+			{
+				return true;
+			}
+
+			bool shouldStop =
+				Vector3.Distance(serverState.ActiveThrow.OriginWorldPos, serverState.WorldPosition) >= serverState.ActiveThrow.WorldTrajectory.magnitude;
+			//			if ( shouldStop ) {
+			//				Logger.Log( $"Should stop throw: {Vector3.Distance( serverState.ActiveThrow.OriginPos, serverState.WorldPosition )}" +
+			//				           $" >= {trajectory.magnitude}" );
+			//			}
+			return shouldStop;
+		}
+	}
+
+	public void NewtonianMove(Vector2Int direction, float speed = Single.NaN)
+	{
+		PushInternal(direction, true, speed);
+	}
+
+	/// <summary>
+	/// Push this thing in provided direction
+	/// </summary>
+	/// <param name="direction"></param>
+	/// <param name="speed"></param>
+	/// <param name="followMode">flag used when object is following its puller
+	/// (turns on tile snapping and removes player collision check)</param>
+	/// <returns>true if push was successful</returns>
+	[Server]
+	public bool Push(Vector2Int direction, float speed = Single.NaN, bool followMode = false, bool ignorePassable = false)
+	{
+		return PushInternal(direction, isNewtonian: false, speed: speed, followMode: followMode, ignorePassable: ignorePassable);
```

Comment:
Codacy found an issue: [Remove this default value assigned to parameter 'isNewtonian'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)